app/src/main/java/nl/tudelft/b_b_w/blockchain/Block.java

```diff
@@ -0,0 +1,311 @@
+     *
+     * @return hash of sender block
+     */
+    public final Hash getPreviousHashSender() {
+        return data.getPreviousHashSender();
+    }
+
+    /**
+     * This method returns the trust value of the block
+     *
+     * @return int trust value of block.
+     */
+    public final double getTrustValue() {
+        return data.getTrustValue();
+    }
+
+    /**
+     * This method set the trust value of the block
+     *
+     * @param trustValue of the block
+     */
+    public final void setTrustValue(double trustValue) {
+        data.setTrustValue(trustValue);
+    }
+
+
+    /**************************************END******************************************/
+
+    /**
+     * This method returns the hash of the block
+     *
+     * @return Hash object of the block's hash
+     */
+    public final Hash getOwnHash() {
+        return ownHash;
+    }
+
+    /**
+     * Calculate the SHA-256 hash of this block
+     *
+     * @return the base-64 encoded hash as a string
+     */
+    private Hash calculateHash() {
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA-256");
+            String text = this.getOwnerName()
+                    + this.getOwnerIban()
+                    + this.getOwnerPublicKey().toString()
+                    + this.getContactName()
+                    + this.getContactIban()
+                    + this.getContactPublicKey().toString()
+                    + this.getBlockType().name()
+                    + this.getSequenceNumber()
+                    + this.getPreviousHashChain().toString()
+                    + this.getPreviousHashSender().toString();
+            md.update(text.getBytes("UTF-8"));
+            byte[] digest = md.digest();
+            Hash hash = new Hash(String.format("%064x", new BigInteger(1, digest)));
+            return hash;
+        } catch (Exception e) {
+            return new Hash(e.getMessage());
+        }
+    }
+
+
+    /**
+     * Attributes are isRevoked, iban and public key
+     *
+     * @param o given block
+     * @return equals or not
+     */
+    public boolean verifyBlock(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Block block = (Block) o;
+
+        if (isRevoked() != block.isRevoked()) {
+            return false;
+        }
+        if (!getContactIban().equals(block.getContactIban())) {
+            return false;
+        }
+        return getContactPublicKey().equals(block.getContactPublicKey());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Block)) {
+            return false;
+        }
+
+        Block block = (Block) o;
+
+        return ownHash.equals(block.ownHash);
+
```

Comment:
The equals method should work seperately from the hash. If something is changed in some way, after the calculation of the hash, this function will return true, while this should not be the case.